presqt/api_v1/utilities/multiprocess/watchdog.py

```diff
@@ -0,0 +1,41 @@
+from time import sleep
+
+from presqt.api_v1.utilities import write_file
+from presqt.api_v1.utilities.io.read_file import read_file
+
+
+def process_watchdog(process_function, process_info_path, process_time, process_state):
+    """
+    Watchdog for file transfer processes spawned off using Multiprocessing. It will monitor if the
+    process has either finished or has gone over it's processing time.
+
+    Parameters
+    ----------
+    process_function : class
```

Comment:
When you type(process_function) does it come back with `class`?

Reply:
<class 'multiprocessing.context.Process'>